src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt

```diff
@@ -18,17 +18,37 @@ class DemoController {
     @Synchronized
     private fun generateFileName(): String = UUID.randomUUID().toString()
 
-    private fun getDemoFile(): File = File("demo-${generateFileName()}.kt")
+    private fun getDemoFile(generatedName: String = generateFileName()): File = File("demo-$generatedName.kt")
+    private fun getDemoConfig(generatedName: String = generateFileName()): File = File("diktat-analysis-$generatedName.yml")
+
+    /**
+     * @param configLines [String] of yaml config
+     * @param generatedName autogenerated name
+     * @return temp [File] that contains yaml config
+     */
+    fun prepareConfigFile(
+        configLines: String,
+        generatedName: String = generateFileName(),
+    ) = getDemoConfig(generatedName).apply {
+        writeText(configLines)
+    }
 
     /**
      * @param codeFormHtml
      * @return a page name
      */
+    @Suppress("TOO_LONG_FUNCTION")
     fun checkAndFixCode(codeFormHtml: CodeForm): CodeForm {
         val codeForm = codeFormHtml
-        val codeFix = CodeFix(codeForm.initialCode!!, codeFormHtml.ruleSet[0])
-        val file = getDemoFile()
+        val generatedName = generateFileName()
+        println(codeFormHtml.diktatConfig)
```

Comment:
is it okay to have println here? (maybe left debug logs)

Reply:
Sure it is. Will fix it